sys/net/application_layer/nanocoap/nanocoap.c

```diff
@@ -513,6 +513,16 @@ ssize_t coap_build_reply(coap_pkt_t *pkt, unsigned code,
         }
     }
 
+    uint32_t no_response;
+    if (coap_opt_get_uint(pkt, COAP_OPT_NO_RESPONSE, &no_response) == 0) {
+        uint8_t mask = 1 << (code >> 6);
```

Comment:
```suggestion
        uint8_t no_response_index = (code >> 5) - 1;
        /* If the handler code misbehaved here, we'd face UB otherise */
        assert(no_response_index < 7);
        uint8_t mask = 1 << no_response_index;
```

Calculation is not correct -- it's "take the major part of the response code, and reduce that by 1 because 0 would be a request code". (Found this when testing with the nanocoap_server example, which would have sent the 4.04 response even though No-Response:26(ever) was sent)